clang/lib/Serialization/ASTWriter.cpp

```diff
@@ -5706,6 +5707,27 @@ void ASTWriter::WriteDeclAndTypes(ASTContext &Context) {
     Stream.EmitRecord(DELAYED_NAMESPACE_LEXICAL_VISIBLE_RECORD,
                       DelayedNamespaceRecord);
 
+  if (!FunctionToLambdasMap.empty()) {
+    // TODO: on disk hash table for function to lambda mapping might be more
+    // efficent becuase it allows lazy deserialization.
+    RecordData FunctionToLambdasMapRecord;
+    for (const auto &Pair : FunctionToLambdasMap) {
```

Comment:
This looks like the non-deterministic for loop, this is the DenseMap iteration that decides what to write to disk. I'll just send a patch to make this a MapVector.

Reply:
Nevermind, I see you have a better fix.